crates/nu-command/src/filesystem/du.rs

```diff
@@ -93,81 +96,69 @@ impl Command for Du {
                 });
             }
         }
+        let all = call.has_flag(engine_state, stack, "all")?;
+        let deref = call.has_flag(engine_state, stack, "deref")?;
+        let exclude = call.get_flag(engine_state, stack, "exclude")?;
         let current_dir = current_dir(engine_state, stack)?;
 
-        let args = DuArgs {
-            path: opt_for_glob_pattern(engine_state, stack, call, 0)?,
-            all: call.has_flag(engine_state, stack, "all")?,
-            deref: call.has_flag(engine_state, stack, "deref")?,
-            exclude: call.get_flag(engine_state, stack, "exclude")?,
-            max_depth,
-            min_size,
-        };
-
-        let exclude = args.exclude.map_or(Ok(None), move |x| {
-            Pattern::new(x.item.as_ref())
-                .map(Some)
-                .map_err(|e| ShellError::InvalidGlobPattern {
-                    msg: e.msg.into(),
-                    span: x.span,
-                })
-        })?;
-
-        let include_files = args.all;
-        let mut paths = match args.path {
-            Some(p) => nu_engine::glob_from(&p, &current_dir, call.head, None),
-            // The * pattern should never fail.
-            None => nu_engine::glob_from(
-                &Spanned {
-                    item: NuGlob::Expand("*".into()),
-                    span: Span::unknown(),
-                },
-                &current_dir,
-                call.head,
-                None,
-            ),
-        }
-        .map(|f| f.1)?
-        .filter(move |p| {
-            if include_files {
-                true
-            } else {
-                matches!(p, Ok(f) if f.is_dir())
+        let paths = get_rest_for_glob_pattern(engine_state, stack, call, 0)?;
+        let paths = if paths.is_empty() { None } else { Some(paths) };
```

Comment:
I wonder if we should check `call.positional_len()` to determine whether `ls` should list the current directory.

I.e., should `ls ...[]` list nothing or should it list the current directory.

(Some thing goes for `du`)

Reply:
Hmm...I expect `ls ...[]` returns empty.
But in the case, `call.positional_len()` is also `0`

Reply:
I figure out a way to do it properly: `call.arguments.iter().count() == 0`.
But I'm not really sure if it's good to use `call.arguments` directly

Reply:
> But in the case, `call.positional_len()` is also `0`

Oh, that's right.

> call.arguments.iter().count() == 0

I think this will also include flags in the argument count.

Rather, I think `call.rest_iter().count()` will work? If not, it should be fine to use `call.arguments` directly.

Reply:
I test `rest_iter().count()` works as expected :)